### PR TITLE
HPCC-32111 Use compression in global merge activity

### PR DESCRIPTION
**Pull request by jakesmith**
## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [x] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by jakesmith** [comment]
@ghalliday - please review. Questionable whether should be 9.8 vs master.

File: thorlcr/activities/merge/thmergeslave.cpp [hunk 1]
```diff
@@ -270,6 +271,16 @@ class GlobalMergeSlaveActivity : public CSlaveActivity
...
         masterMpTag = container.queryJobChannel().deserializeMPTag(data);
+        rwFlags = DEFAULT_RWFLAGS;
+        if (getOptBool(THOROPT_COMPRESS_SPILLS, true))
+        {
+            StringBuffer compType;
+            getOpt(THOROPT_COMPRESS_SPILL_TYPE, compType);
+            unsigned spillCompInfo;
+            setCompFlag(compType, spillCompInfo);
+            if (spillCompInfo)
+                rwFlags |= rw_compress;
```
Comment (ghalliday) [inline thorlcr/activities/merge/thmergeslave.cpp:282]: rwFlags |= spillCompInfo;

for consistency with elsewhere?  Really this should be a single piece of common code.... but that can be a separate refactoring.  The same logic seems to be done two different ways.

**Review by ghalliday — COMMENTED** [review]
@jakesmith 1 minor question.

File: thorlcr/activities/merge/thmergeslave.cpp (hunk 1 above)
Comment (jakesmith) [reply thorlcr/activities/merge/thmergeslave.cpp]: Yes, I meant to have it like that, now fixed.
Have opened separate JIRA to common code like this: https://hpccsystems.atlassian.net/browse/HPCC-32881

**Review by ghalliday — APPROVED** [review]

**Comment by jakesmith** [comment]
@ghalliday - uninitialized issue fixed/squashed in.

**Review by ghalliday — APPROVED** [review]

**Pull request merged**